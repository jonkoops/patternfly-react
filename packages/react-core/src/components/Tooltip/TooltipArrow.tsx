import { HTMLProps, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/Tooltip/tooltip';
import { css } from '@patternfly/react-styles';

export interface TooltipArrowProps extends HTMLProps<HTMLDivElement> {
  /** className */
  className?: string;
}

export const TooltipArrow: FunctionComponent<TooltipArrowProps> = ({ className, ...props }: TooltipArrowProps) => (
  <div className={css(styles.tooltipArrow, className)} {...props} />
);
TooltipArrow.displayName = 'TooltipArrow';
