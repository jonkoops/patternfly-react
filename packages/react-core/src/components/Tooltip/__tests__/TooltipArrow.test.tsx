/**
 * @vitest-environment jsdom
 */
import styles from '@patternfly/react-styles/css/components/Tooltip/tooltip';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { expect, test } from 'vitest';

import { TooltipArrow } from '../TooltipArrow';

test('Renders without children', () => {
  render(<TooltipArrow data-testid="tooltipArrow" />);

  expect(screen.getByTestId('tooltipArrow')).toBeVisible();
});

test('Renders with class name pf-v5-c-tooltip__arrow by default', () => {
  render(<TooltipArrow data-testid="tooltipArrow" />);

  expect(screen.getByTestId('tooltipArrow')).toHaveClass(styles.tooltipArrow);
});

test('Renders with custom class names provided via prop', () => {
  render(<TooltipArrow className="test-class" data-testid="tooltipArrow" />);

  expect(screen.getByTestId('tooltipArrow')).toHaveClass('test-class');
});

test('Matches the snapshot', () => {
  const { asFragment } = render(<TooltipArrow data-testid="tooltipArrow" />);
  expect(asFragment()).toMatchSnapshot();
});
