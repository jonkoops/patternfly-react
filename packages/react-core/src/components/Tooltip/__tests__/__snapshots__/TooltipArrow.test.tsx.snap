// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-tooltip__arrow"
    data-testid="tooltipArrow"
  />
</DocumentFragment>
`;
